lms/djangoapps/grades/api/v1/views.py

```diff
@@ -0,0 +1,192 @@
+        except InvalidKeyError:
+            return self.make_error_response(
+                status_code=status.HTTP_404_NOT_FOUND,
+                developer_message='The provided course key cannot be parsed.',
+                error_code='invalid_course_key'
+            )
+
+        if not CourseOverview.get_from_id_if_exists(course_key):
+            return self.make_error_response(
+                status_code=status.HTTP_404_NOT_FOUND,
+                developer_message="Requested grade for unknown course {course}".format(course=course_id),
+                error_code='course_does_not_exist'
+            )
+
+        return course_key
+
+    def _get_effective_user(self, request, course_key):
+        """
+        Returns the user object corresponding to the request's 'username' parameter,
+        or the current request.user if no 'username' was provided.
+        """
+        if 'username' in request.GET:
+            username = request.GET.get('username')
+        else:
+            username = request.user.username
+
+        try:
+            user = USER_MODEL.objects.get(username=username)
+        except USER_MODEL.DoesNotExist:
+            return self.make_error_response(
+                status_code=status.HTTP_404_NOT_FOUND,
+                developer_message='The user matching the requested username does not exist.',
+                error_code='user_does_not_exist'
+            )
+
+        if not enrollment_data.get_course_enrollment(username, str(course_key)):
+            return self.make_error_response(
+                status_code=status.HTTP_404_NOT_FOUND,
+                developer_message='The user matching the requested username is not enrolled in this course',
+                error_code='user_not_enrolled'
+            )
+
+        return user
+
+    def _make_grade_response(self, user, course_key, course_grade):
+        """
+        Serialize a single grade to dict to use in Responses
+        """
+        return {
+            'username': user.username,
+            'email': user.email,
+            'course_id': str(course_key),
+            'passed': course_grade.passed,
+            'percent': course_grade.percent,
+            'letter_grade': course_grade.letter_grade,
+        }
+
+    def perform_authentication(self, request):
+        """
+        Ensures that the user is authenticated (e.g. not an AnonymousUser), unless DEBUG mode is enabled.
+        """
+        super(GradeViewMixin, self).perform_authentication(request)
+        if request.user.is_anonymous():
+            raise AuthenticationFailed
+
+
+class CourseGradesView(GradeViewMixin, GenericAPIView):
+    """
+    **Use Case**
+        * Get course grades if all user who are enrolled in a course.
+        The currently logged-in user may request all enrolled user's grades information.
+    **Example Request**
+        GET /api/grades/v1/courses/{course_id}/   - Get grades for all users in course
```

Comment:
Please include both example requests - including the one where `course_id` is passed as a filter via a `GET` parameter.